compiler/rustc_lint_defs/src/builtin.rs

```diff
@@ -934,11 +933,8 @@ declare_lint! {
     /// with FFI interfaces or complex type aliases, where it triggers
     /// incorrectly, or in situations where it will be more difficult to
     /// clearly express the intent. It may be possible that this will become a
-    /// warning in the future, possibly with [type ascription] providing a
-    /// convenient way to work around the current issues. See [RFC 401] for
-    /// historical context.
+    /// warning in the future. See [RFC 401] for historical context.
```

Comment:
I don't think we should say that this may become a warning in the future without a reason. I'll leave it to you to decide whether we should remove the sentence about the warning (after all, we are free to raise any lints to warn however we choose, removing this doesn't make that impossible in the future) or whether we should keep a mention of type ascription (as that feature may still be added in the future with good syntax).

Reply:
I've put back that we might add syntax for coercions without calling it "type ascription" and moved type ascription to the "historical context" section instead. My main motivation here is to deter people from using type ascription in its current form, especially in the stable documentation.